### PR TITLE
Changed REST client to MP Rest Client

### PR DESCRIPTION
**Pull request by justineechen**
- Changed Client to use Microprofile Rest Client.
- Moved private methods from InventoryManager into InventoryUtils to avoid confusion for the user when they are going through the guide.

### CODE REVIEW

**Comment by evelinec** [comment]
@justineechen pls take action on the above ^

**Comment by JunyiSun** [comment]
@justineechen You added files to the `start` directory that are supposed to be added by the reader according to our guide.  Please remove those extra content and keep them unchanged. EX) InventoryResource.java, and a few config files.

**Comment by evelinec** [comment]
@JunyiSun can you review the latest commits pls? Thanks.

File: finish/src/main/liberty/config/server.xml [hunk 1]
```diff
@@ -4,9 +4,7 @@
     <feature>jaxrs-2.0</feature>
     <feature>jsonp-1.0</feature>
     <feature>cdi-1.2</feature>
-    <!-- tag::config[] -->
-    <feature>mpConfig-1.2</feature>
-    <!-- end::config[] -->
```
Comment (JunyiSun) [inline finish/src/main/liberty/config/server.xml]: I already remove the extra config feature last time, so please don't remove this one.

File: start/src/main/liberty/config/server.xml [hunk 2]
```diff
@@ -4,7 +4,7 @@
     <feature>jaxrs-2.0</feature>
     <feature>jsonp-1.0</feature>
     <feature>cdi-1.2</feature>
-    <feature>mpConfig-1.2</feature>
```
Comment (JunyiSun) [inline start/src/main/liberty/config/server.xml]: Please don't remove the only config feature here, I removed the extra one last time already

File: start/src/test/java/it/io/openliberty/guides/inventory/InventoryEndpointTest.java [hunk 3]
```diff
@@ -76,7 +76,7 @@ public void testEmptyInventory() {
     int expected = 0;
     int actual = obj.getInt("total");
     assertEquals("The inventory should be empty on application start but it wasn't",
-                 expected, actual);
```
Comment (JunyiSun) [inline start/src/test/java/it/io/openliberty/guides/inventory/InventoryEndpointTest.java]: Ideally, if you don't make any code changes to the existing code, don't re-format them. The existing code are formatted this way to look good on the website.

**Pull request merged**